src/main/java/com/google/sps/model/GeocodingClient.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.maps.model.AddressType;
+import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceType;
+import com.google.sps.exceptions.GeocodingException;
+
+/**
+ * Contract for sending GET requests to the Google Geocoding API. Implement getGeocodingResult to
+ * obtain the corresponding GeocodingResult of an address.
+ */
+public interface GeocodingClient {
+  /**
+   * Sends a GET request to the Google Geocoding API to convert from address to GeocodingResult.
+   *
+   * @param address A String representing the address to geocode.
+   * @return A GeocodingResult returned from the Geocoding API.
+   * @throws GeocodingException A custom exception is thrown to signal an error pertaining to the
+   *     Geocoding API.
+   */
+  GeocodingResult getGeocodingResult(String address) throws GeocodingException;
+
+  /**
+   * Parses for coordinates in a resulting call to the Geocoding API.
+   *
+   * @param result A GeocodingResult returned from the Geocoding API.
+   * @return A LatLng representing coordinates.
+   */
+  public static LatLng getCoordinates(GeocodingResult result) {
+    return result.geometry.location;
+  }
+
+  /**
+   * Converts an AddressType to a PlaceType for the purpose of calling the Places API. PlaceType is
+   * a subset of AddressType and hence, not all AddressTypes are supported by the Places API.
+   *
+   * @param addressType An AddressType to convert to a PlaceType
+   * @return A PlaceType corresponding to an AddressType if available, null if not available.
+   */
+  public static PlaceType convertAddressTypeToPlaceType(AddressType addressType) {
```

Comment:
nit: Use an Optional here if null is a possibility

Reply:
I don't know if that's the best use for [Optional](https://medium.com/faun/working-on-null-elegantly-with-java-optional-62f5e65869c5#:~:text=Optional%20is%20primarily%20intended%20for,point%20to%20an%20Optional%20instance.). I could see how it is useful if you're going to access the returned object to avoid a NullPointerException but this method returns a PlaceType enum and nulls are filtered shortly after in the Go servlet where this is used. I've also added a sentence to the javadoc to somewhat reflect this if that helps.

Reply:
I think this would be a great use of `Optional`, because you expect to have address types that don't map to a place type, and there is no natural way to represent "not a valid type" using `PlaceType` itself.

(Sometimes enums will have some sort of UNKNOWN value you could use for this, but I don't think there is such a place type.)

Reply:
Oh okay, so it would be used in place of `PlaceType.UNKNOWN` instead of what the article I linked suggested to use Optional for i.e. avoiding null check validation. I think that makes sense, thanks!

(There's one for AddressType but unfortunately not for PlaceType)

Reply:
`RESOLVED`